src/MediatR/IMediator.cs

```diff
@@ -25,5 +26,13 @@ public interface IMediator
         /// <returns>A task that represents the publish operation.</returns>
         Task Publish<TNotification>(TNotification notification, CancellationToken cancellationToken = default)
             where TNotification : INotification;
+        /// <summary>
+        /// Asynchronously send a notification to multiple handlers
+        /// </summary>
+        /// <param name="notification">Notification object</param>
+        /// <param name="notificationType">Type of notification object</param>
+        /// <param name="cancellationToken">Optional cancellation token</param>
+        /// <returns>A task that represents the publish operation.</returns>
+        Task Publish(INotification notification, Type notificationType, CancellationToken cancellationToken = default);
```

Comment:
Can't you infer the notification type from the notification itself? And I would assume that the weakly-typed would mean `object`, not `INotification`.

If you look at the `Send` method, it infers the request type from the request object passed in. I would expect this method to look like:

```
Task Publish(object notification, CancellationToken cancellationToken = default);
```

That way it's very obvious what you're trying to do here. I think callers wouldn't want to have to call `GetType` before publishing.